engine/src/configuration/applier/contactgroup.cc

```diff
@@ -263,3 +431,49 @@ void applier::contactgroup::_resolve_members(
     }
   }
 }
+#else
+/**
+ * @brief Resolve members of a contact group. A contact group can be defined
+ * from others contactgroups. But we only want for engine, contactgroups defined
+ * with contacts. So if it contains contactgroups, we have to copy their
+ * contacts into this contactgroup and then empty the contactgroups members.
+ *
+ * @param s  Configuration being applied.
+ * @param obj Object that should be processed.
+ * @param resolved a reference to keep a trace of already expanded
+ * contactgroups.
+ */
+void applier::contactgroup::_resolve_members(
+    configuration::State& s,
+    configuration::Contactgroup& obj,
+    absl::flat_hash_set<std::string_view>& resolved) {
+  if (resolved.contains(obj.contactgroup_name()))
+    return;
+
+  resolved.emplace(obj.contactgroup_name());
+  if (!obj.contactgroup_members().data().empty()) {
+    // Logging.
+    config_logger->debug("Resolving members of contact group '{}'",
+                  obj.contactgroup_name());
+    for (auto& cg_name : obj.contactgroup_members().data()) {
+      auto it = std::find_if(s.mutable_contactgroups()->begin(),
```

Comment:
why not store this members in an strict order. By do this, you could avoid to scan all container and do a binary_search instead

Reply:
The strict order is difficult to acquieve since the configuration comes from a cfg file. And also protobuf doesn't provide set. When the configuration will be full protobuf and only read by broker, there will be possible improvements I think.